packages/web3-core/src/types.ts

```diff
@@ -40,6 +43,16 @@ export type Web3BaseProviderConstructor = new <API extends Web3APISpec>(
 	net?: Socket,
 ) => Web3BaseProvider<API>;
 
+// TODO: When we have `web3-types` package we can share TransactionType
+export type TransactionTypeParser = (transaction: Record<string, unknown>) => HexString | undefined;
+
+// TODO: When we have `web3-types` package we can share TransactionType
+export type TransactionBuilder = (options: {
+	transaction: Record<string, unknown>;
+	web3Context: Web3Context<EthExecutionAPI>;
```

Comment:
Also `TransactionBuilder` should support different APIs being passed e.g. `Web3EthApi`

Reply:
Ok